src/GPS.cpp

```diff
@@ -0,0 +1,120 @@
+#include "GPS.hpp"
+#include <nmea0183/GPS.hpp>
+
+using namespace nmea0183;
+using namespace marnav;
+using namespace std;
+using namespace gps_base;
+
+GPS_SOLUTION_TYPES GPS::getPositionType(utils::optional<nmea::mode_indicator> mode)
```

Comment:
Force the check on non-optional values as well
```suggestion
GPS_SOLUTION_TYPES GPS::getPositionType(nmea::mode_indicator mode)
```